eclass/java-pkg-simple.eclass

```diff
@@ -415,22 +415,25 @@ java-pkg-simple_src_compile() {
 	fi
 
 	# package
-	[[ ! -d ${classes}/META-INF ]] && mkdir ${classes}/META-INF
-	if  [[ -v JAVA_AUTOMATIC_MODULE_NAME ]]; then
-		echo "Automatic-Module-Name: ${JAVA_AUTOMATIC_MODULE_NAME}" \
-			>> ${classes}/META-INF/MANIFEST.MF || die "adding module name failed"
-	fi
-	if  [[ -v JAVA_MAIN_CLASS ]]; then
-		echo "Main-Class: ${JAVA_MAIN_CLASS}" \
-			>> ${classes}/META-INF/MANIFEST.MF || die "adding main class failed"
-	fi
 	local jar_args
 	if [[ -e ${classes}/META-INF/MANIFEST.MF ]]; then
 		jar_args="cfm ${JAVA_JAR_FILENAME} ${classes}/META-INF/MANIFEST.MF"
 	else
 		jar_args="cf ${JAVA_JAR_FILENAME}"
 	fi
 	jar ${jar_args} -C ${classes} . || die "jar failed"
+	if  [[ -v JAVA_AUTOMATIC_MODULE_NAME ]]; then
+		echo "Automatic-Module-Name: ${JAVA_AUTOMATIC_MODULE_NAME}" \
+			>> "${T}/add-to-MANIFEST.MF" || die "adding module name failed"
+	fi
+	if  [[ -v JAVA_MAIN_CLASS ]]; then
+		echo "Main-Class: ${JAVA_MAIN_CLASS}" \
+			>> "${T}/add-to-MANIFEST.MF" || die "adding main class failed"
+	fi
+	if [[ -e ${T}/add-to-MANIFEST.MF ]]; then
```

Comment:
quote T and use -f:
```suggestion
	if [[ -f "${T}"/add-to-MANIFEST.MF ]]; then
```

Reply:
Thanks for the proposal. But this way we would have to do it twice: once for Main-Class and once for Automatic-module-name.
I found another solution checking for non-zero length and deleting the `add-to-MANIFEST.MF` file at the end